src/link_prober/LinkProber.cpp

```diff
@@ -232,18 +232,18 @@ void LinkProber::handleUpdateEthernetFrame()
 //
 void LinkProber::handleSendSwitchCommand()
 {
-    IcmpPayload *icmpPayload = reinterpret_cast<IcmpPayload *> (
-        mTxBuffer.data() + sizeof(ether_header) + sizeof(iphdr) + sizeof(icmphdr)
-    );
-    icmpPayload->command = htonl(static_cast<uint32_t> (Command::COMMAND_SWITCH_ACTIVE));
+    new (mTxBuffer.data() + sizeof(ether_header) + sizeof(iphdr) + sizeof(icmphdr) + sizeof(IcmpPayload)) TlvCommand();
```

Comment:
I suggest adding another variable like `icmp_payload_offset = sizeof(ether_header) + sizeof(iphdr) + sizeof(icmphdr) + sizeof(IcmpPayload)` so that we don't have to do same calculation each time.